osidb/tests/test_flaw.py

```diff
@@ -725,3 +726,55 @@ def test_validate_affect_ps_module_errors(self, bz_id, ps_module, should_raise):
         else:
             affect = AffectFactory(flaw=flaw, ps_module=ps_module)
             assert affect
+
+    @pytest.mark.parametrize(
+        "group,unembargo_date,should_raise",
+        [
+            (settings.PUBLIC_READ_GROUPS, None, True),
+            (settings.PUBLIC_READ_GROUPS, tzdatetime(2000, 1, 1), False),
+            (settings.EMBARGO_READ_GROUP, None, False),
+            (settings.EMBARGO_READ_GROUP, tzdatetime(2000, 1, 1), False),
+        ],
+    )
+    def test_validate_unembargo_date_public(self, group, unembargo_date, should_raise):
+        if should_raise:
+            with pytest.raises(ValidationError) as e:
+                flaw = Flaw.objects.create(
+                    title="title",
+                    description="description",
+                    reported_dt=tzdatetime(2000, 1, 1),
+                    unembargo_dt=unembargo_date,
+                    acl_read=[
+                        uuid.uuid5(
+                            uuid.NAMESPACE_URL,
+                            f"https://osidb.prod.redhat.com/ns/acls#{group}",
+                        )
+                    ],
+                    acl_write=[
+                        uuid.uuid5(
+                            uuid.NAMESPACE_URL,
+                            f"https://osidb.prod.redhat.com/ns/acls#{settings.EMBARGO_WRITE_GROUP}",
+                        )
+                    ],
+                )
```

Comment:
You can use `FlawFactory` instead and pass it `embargoed=True` as an attribute and will correctly set the ACLs IIRC, the benefit is that you'd only have to set `embargoed`, `reported_dt` and `unembargo_dt`.

Reply:
Thank you for the feedback, changed in current PR.